pinot-query-planner/src/main/java/org/apache/pinot/calcite/rel/rules/PinotJoinExchangeNodeInsertRule.java

```diff
@@ -54,10 +54,15 @@ public void onMatch(RelOptRuleCall call) {
     JoinInfo joinInfo = join.analyzeCondition();
     RelNode newLeft;
     RelNode newRight;
-    if (PinotHintOptions.JoinHintOptions.useLookupJoinStrategy(join)) {
+    String joinStrategyHint = PinotHintOptions.JoinHintOptions.getJoinStrategyHint(join);
+    if (PinotHintOptions.JoinHintOptions.useLookupJoinStrategy(joinStrategyHint)) {
```

Comment:
nit: what about having a `PinotHintOptions.JoinHintOptions.getJoinStrategy` that returns an enum we can use in a switch here? We expect to add at least 3 new extra strategies (listed in https://github.com/apache/pinot/issues/14518, including random + broadcast right now cannot be specified), so the switch syntax may be helpful.

Reply:
Done